src/__experimental__/components/date/date.spec.js

```diff
@@ -0,0 +1,333 @@
+import React from 'react';
+import moment from 'moment';
+import { mount } from 'enzyme';
+import Date from './date.component';
+import DatePicker from './date-picker.component';
+import Textbox from '../textbox';
+
+describe('Date', () => {
+  let wrapper;
+
+  describe('when the Component is rendered', () => {
+    it("then the Textbox component should be rendered as it's descendant", () => {
+      wrapper = render({});
+      expect(wrapper.find(Textbox).exists()).toBe(true);
+    });
+  });
+
+  describe('when the "value" prop is an empty string', () => {
+    const currentDate = getFormattedDate(moment());
+
+    it('then the input element value should be set to today', () => {
+      wrapper = render({ value: '' });
+      simulateBlurOnInput(wrapper);
+      expect(wrapper.find('input').prop('value')).toBe(currentDate);
+    });
+  });
+
+  describe('when "autoFocus" prop is defined', () => {
+    it("then component's input should be focused after render", () => {
+      wrapper = render({ autoFocus: true });
+      const input = wrapper.find('input');
+      const focusedElement = document.activeElement;
+      expect(input.instance()).toBe(focusedElement);
+    });
+  });
+
+  describe('when autoFocus prop is not defined', () => {
+    it('does not sets focus on the input', () => {
```

Comment:
Not sure this tests is needed? Does it add any value in terms of coverage?

Reply:
1.Yes, this test is needed, we are expecting the component to behave differently on mount depending on that prop.
2.Yes, coverage also includes `else` conditions.